test/kafka-exactly-once/after-restart.td

```diff
@@ -7,110 +7,98 @@
+        {
+          "name": "data",
+          "type": {
             "type": "record",
+            "name": "data",
             "fields": [
               {"name": "a", "type": "long"},
               {"name": "b", "type": "long"}
             ]
-          },
-          "null"
-        ]
-      },
-      { "name": "after", "type": ["row", "null"] }
-    ]
-  }
-
-$ set trxschemakey={
-      "name": "io.debezium.connector.common.TransactionMetadataKey",
-      "type": "record",
-      "fields": [
-          {
-              "name": "id",
-              "type": "string"
           }
+        },
+        {
+          "name": "time",
+          "type": "long"
+        },
+        {
+          "name": "diff",
+          "type": "long"
+        }
       ]
-  }
-
-$ set trxschema={
-    "type":"record", "name":"TransactionMetadataValue", "namespace":"io.debezium.connector.common",
-    "fields":[
-    {"name":"status","type":"string"},
-    {"name":"id","type": "string"},
-    {"name": "event_count",
-    "type": ["null", "long"],
-    "default": null
-    },
-    {"name":"data_collections","type":["null",{"type":"array",
-    "items": {"type":"record",
-    "name":"ConnectDefault",
-    "namespace":"io.confluent.connect.Avro",
-    "fields": [ {
-    "name": "data_collection",
-    "type": "string"
-    },
-    {
-    "name": "event_count",
-    "type": "long" }]}}],
-    "default": null}],
-    "connect.name": "io.debezium.connector.common.TransactionMetadataValue"
     }
-
-$ kafka-ingest format=avro topic=input-consistency timestamp=1 schema=${trxschemakey}
-{"id": "52"}
-
-$ kafka-ingest format=avro topic=input-consistency timestamp=1 schema=${trxschema}
-{"status":"BEGIN","id":"52","event_count":null,"data_collections":null}
-{"status":"END","id":"52","event_count":{"long": 2},"data_collections":{"array": [{"event_count": 2, "data_collection": "testdrive-input-${testdrive.seed}"}]}}
+  },
+  {
+    "type": "record",
+    "name": "progress",
+    "namespace": "com.materialize.cdc",
+    "fields": [
+      {
+        "name": "lower",
+        "type": {
+          "type": "array",
+          "items": "long"
+        }
+      },
+      {
+        "name": "upper",
+        "type": {
+          "type": "array",
+          "items": "long"
+        }
+      },
+      {
+        "name": "counts",
+        "type": {
+          "type": "array",
+          "items": {
+            "type": "record",
+            "name": "counts",
+            "fields": [
+              {
+                "name": "time",
+                "type": "long"
+              },
+              {
+                "name": "count",
+                "type": "long"
+              }
+            ]
+          }
+        }
+      }
+    ]
+  }
+  ]
 
 $ kafka-ingest format=avro topic=input schema=${schema} timestamp=14
-{"before": null, "after": {"row": {"a": 4, "b": 1}}}
-{"before": null, "after": {"row": {"a": 5, "b": 2}}}
+{"array":[{"data":{"a":4,"b":1},"time":4,"diff":1}]}
```

Comment:
What's the difference between these two timestamps?  Are they in different time domains?

Reply:
Ah, I meant to delete all these. The timestamp passed to `kafka-ingest` doesn't matter in a CDCv2 stream. It's a timestamp that kafka records for each messages you publish. Debezium dedup cares about it, it's what we call in the code `upstream_time_millis`